analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Sonar/SymbolicExecutionAnalyzerFactoryTest.cs

```diff
@@ -88,13 +85,22 @@ public void GetEnabledAnalyzers_ReturnsEmptyList_WhenDiagnosticsAreDisabled()
             var sut = new SymbolicExecutionAnalyzerFactory();
             var diagnostics = symbolicExecutionRuleIds.ToImmutableDictionary(v => v, _ => ReportDiagnostic.Suppress);
             var context = CreateSyntaxNodeAnalysisContext(diagnostics);
-            var analyzers = sut.GetEnabledAnalyzers(context).ToList();
-            var enabledAnalyzers =
-                analyzers
-                    .SelectMany(analyzer => analyzer.SupportedDiagnostics.Select(descriptor => descriptor.Id))
-                    .ToList();
+            var analyzers = sut.GetEnabledAnalyzers(context, false, false).ToList();
+            var enabledAnalyzers = analyzers.SelectMany(analyzer => analyzer.SupportedDiagnostics.Select(descriptor => descriptor.Id)).ToList();
+
+            enabledAnalyzers.Should().BeEmpty();
+        }
+
+        [TestMethod]
+        public void GetEnabledAnalyzers_ReturnsEmptyList_OnTestProjectScannerRun()  // Old SE rules do not run on test code
+        {
+            var sut = new SymbolicExecutionAnalyzerFactory();
+            var diagnostics = symbolicExecutionRuleIds.ToImmutableDictionary(v => v, _ => ReportDiagnostic.Warn);
+            var context = CreateSyntaxNodeAnalysisContext(diagnostics);
+            var analyzers = sut.GetEnabledAnalyzers(context, true, true).ToList();
+            var enabledAnalyzers = analyzers.SelectMany(analyzer => analyzer.SupportedDiagnostics.Select(descriptor => descriptor.Id)).ToList();
 
-            CollectionAssert.AreEquivalent(new List<string>(), enabledAnalyzers);
+            enabledAnalyzers.Should().BeEmpty();
         }
```

Comment:
Please add another test like the one above with the value of false for the isScannerRun parameter. That will check that the "All" scoped rules are run in the Nuget case.

Reply:
The entire `SymbolicExecutionAnalyzerFactory` and `SymbolicExecutionAnalyzerFactoryTest` will be deleted quite soon. All these various scenarios will be tested in #5159